kafka-message-signing/src/main/kotlin/com/gxf/utilities/kafka/message/wrapper/SignableMessageWrapper.kt

```diff
@@ -5,19 +5,28 @@ package com.gxf.utilities.kafka.message.wrapper
 
 import java.io.IOException
 import java.nio.ByteBuffer
+import java.util.function.Consumer
+import java.util.function.Function
 
 /**
  * Wrapper for signable messages. Because these messages are generated from Avro schemas, they can't be changed. This
  * wrapper unifies them for the MessageSigner.
  */
-abstract class SignableMessageWrapper<T>(val message: T) {
+class SignableMessageWrapper<T>(
+    val message: T,
+    private val messageGetter: Function<T, ByteBuffer>,
```

Comment:
I would not use the java.util.function interfaces but the Kotlin syntax (`(T) -> ByteBuffer` in this case)

Reply:
It can then be invoked using `.invoke(..)` or just `(...)`